std/algorithm/iteration.d

```diff
@@ -1034,6 +1049,96 @@ template each(alias pred = "a")
     assert(res == [9, 12, 15]);
 }
 
+// #15357: `each` should behave similar to foreach
+/// each should work with index
+@safe unittest
+{
+    import std.range : iota, lockstep;
```

Comment:
I think you also need to import `sum`, even though it's in same module, so the unittest can be run in the online documentation.